Fixed Font Scaling issue when FontSize is not set explicitly

dipeshmsft opened the pull request:
Fixes #9192  <!-- Issue Number -->

## Description
Fluent styles use 14 as the default size and so in order to handle this we check if the Fluent theme is enabled at the Application level and in that case we used to return the size 14 without scaling it. Hence, even though Text Scaling is set to something other than original size, we did not adjust the font size to match the scaling.

In this PR, I have replaced the DefaultFluentFontSize with DefaultFluentFontSizeFactor and use this to modify the system font size that we get from SystemParameters

<!-- Give a brief summary of the issue and how the pull request is fixing it. -->

## Customer Impact
Developers and users that use scaling other than 100% will see the text in scaled size ( whenever the size is not explicitly set on the control )
<!-- What is the impact to customers of not taking this fix? -->

## Regression
Technically no, as Fluent style is a new feature.
<!-- Is this fixing a problem that was introduced in the most recent release, ie., fixing a regression? -->

## Testing
Local Ad-Hoc testing with sample applications
<!-- What kind of testing has been done with the fix. -->

## Risk
Minimal. No one is using this currently ( until we count the .NET 9 Preview releases )
<!-- Please assess the risk of taking this fix. Provide details backing up your assessment. -->

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/wpf/pull/9719)

h3xds1nz commented on src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/SystemFonts.cs:
```diff
@@ -1023,6 +1023,22 @@ private static double ConvertFontHeight(int height)
+
+            if (dpi != 0)
+            {
+                return (double)(Math.Abs(height) * 96 / dpi);
```
The cast here does nothing. Same as it does nothing in the original function you've copied.

The original function returns `double` but the calculations are done in `int`, so it is truncated and then returned as double. Yours does the computations in double.

dipeshmsft replied:
Removed this

h3xds1nz replied:
Yeah, looks better now.

Though not for this PR, but I wonder about the original intention, the truncation is obviously not ideal for scaling but then again, it's been there since the beginning (even for NetFX).

miloush commented:
Wait a minute I missed the original code too, are you changing what `SPI_NONCLIENTMETRICS` returns based on the theme? I don't think that's very good. Why is this not dealt with in the theme?

dipeshmsft commented:
@miloush, if we handled it in the theme, then the inheritence of FontSize would not have worked as the ThemeStyles \ Styles have priority over inheritence. So, I wanted to handle it at the lowest level possible.

dipeshmsft commented on src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/SystemFonts.cs:
```diff
@@ -428,7 +428,7 @@ public static double MessageFontSize
                 // TODO : Find a better solution to this. Difference in default size of font in Fluent and other themes.
```
I don't think this is the right way to handle this, so I think we should keep the comment here.

h3xds1nz replied:
Yeah if we're not solving it here, then for sure haha.

miloush commented:
I am not quite sure I understand the scenario that doesn't work, but even if that was the case, can you not have this property in the theme assembly and read it from there? Where is this used?

miloush commented:
The answer to where it is used is

https://github.com/dotnet/wpf/blob/4b83839af3c63cda0890f628c83ffb3deef9d06b/src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Documents/TextElement.cs#L643-L652

dipeshmsft commented:
I have modified the above PR to revert the changes in MessageFontSize and introduced an internal property which does the Fluent theme enabled check and returns font size accordingly. This way, the original API is unchanged now, and will enable font scaling for controls when using Fluent themes.

harshit7962 commented on src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/SystemFonts.cs:
```diff
@@ -422,15 +422,24 @@ public static TextDecorationCollection StatusFontTextDecorations
                 if(ThemeManager.IsFluentThemeEnabled)
                 {
-                    return ThemeManager.DefaultFluentThemeFontSize;
+                    return MessageFontSize * ThemeManager.DefaultFluentFontSizeFactor ;
```
Is there a reason we are not doing this in `MessageFontSize` itself? I can think of a case where someone is using the above public property and using Fluent. This will still posses the scaling issues there.

miloush replied:
Yes because `MessageFontSize` is documented to return `SPI_GETNONCLIENTMETRICS` value.

h3xds1nz replied:
As @miloush says, we've raised this in previous comments. Those properties are representing `SystemParametersInfo`, besides being adjusted for DPI, it is partially a raw abstraction that shouldn't change based on active theme.

A new API (e.g. property) would be suitable if we want to return it publicly imho.

miloush replied:
On top of that, we should avoid mixing theme-specific code and logic into non-theme assemblies.

dipeshmsft commented:
/backport to release/9.0

h3xds1nz commented on src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ThemeManager.cs:
```diff
@@ -204,6 +204,11 @@ internal static ResourceDictionary GetThemeDictionary(ThemeMode themeMode)
         return rd;
     }
 
+    internal static bool IsFluentThemeDictionayIncluded()
```
Dictionary*
